application/pages/modplatform/flame/FlameModel.cpp

```diff
@@ -210,79 +210,17 @@ void Flame::ListModel::searchRequestFinished()
-        if (pack.addonId == 0) {
-            qWarning() << "Pack without an ID, skipping: " << pack.name;
-            continue;
-        }
-        pack.name = project.value("name").toString();
-        pack.websiteUrl = project.value("websiteUrl").toString();
-        pack.description = project.value("summary").toString();
-        bool thumbnailFound = false;
-        auto attachments = project.value("attachments").toArray();
-        for(auto attachmentIter: attachments) {
-            auto attachment = attachmentIter.toObject();
-            bool isDefault = attachment.value("isDefault").toBool(false);
-            if(isDefault) {
-                thumbnailFound = true;
-                pack.logoName = attachment.value("title").toString();
-                pack.logoUrl = attachment.value("thumbnailUrl").toString();
-                break;
-            }
-        }
-        if(!thumbnailFound) {
-            qWarning() << "Pack without an icon, skipping: " << pack.name;
-            continue;
-        }
-        auto authors = project.value("authors").toArray();
-        for(auto authorIter: authors) {
-            auto author = authorIter.toObject();
-            ModpackAuthor packAuthor;
-            packAuthor.name = author.value("name").toString();
-            packAuthor.url = author.value("url").toString();
-            pack.authors.append(packAuthor);
-        }
-        int defaultFileId = project.value("defaultFileId").toInt(0);
-        if(defaultFileId == 0) {
-            qWarning() << "Pack without default file, skipping: " << pack.name;
-            continue;
-        }
-        bool found = false;
-        auto files = project.value("latestFiles").toArray();
-        for(auto fileIter: files) {
-            auto file = fileIter.toObject();
-            int id = file.value("id").toInt(0);
-            // NOTE: for now, ignore everything that's not the default...
-            if(id != defaultFileId) {
-                continue;
-            }
-            pack.latestFile.addonId = pack.addonId;
-            pack.latestFile.fileId = id;
-            auto versionArray = file.value("gameVersion").toArray();
-            if(versionArray.size() < 1) {
-                continue;
-            }
-
-            // pick the latest version supported
-            pack.latestFile.mcVersion = versionArray[0].toString();
-            pack.latestFile.version = file.value("displayName").toString();
-            pack.latestFile.downloadUrl = file.value("downloadUrl").toString();
-            found = true;
-            break;
-        }
-        if(!found) {
-            qWarning() << "Pack with no good file, skipping: " << pack.name;
-            continue;
-        }
-        pack.broken = false;
+    QList<Flame::IndexedPack> newList;
+    auto packs = doc.array();
+    for(auto packRaw : packs) {
+        auto packObj = packRaw.toObject();
+
+        Flame::IndexedPack pack;
+        Flame::loadIndexedPack(pack, packObj);
```

Comment:
This call can throw an exception, so should be caught here :)

Here is what I did for modpacks.ch:
```cpp
    ModpacksCH::Modpack pack;
    try
    {
        ModpacksCH::loadModpack(pack, obj);
    }
    catch (const JSONValidationError &e)
    {
        qDebug() << QString::fromUtf8(response);
        qWarning() << "Error while reading pack manifest from FTB: " << e.cause();
        return;
    }
```

Reply:
ah yup, some of the AtLauncher stuff I was copying is missing try/catch apparently